airbyte-cdk/python/airbyte_cdk/test/http/mocker.py

```diff
@@ -0,0 +1,84 @@
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+
+import contextlib
+import functools
+from types import TracebackType
+from typing import Callable, List, Optional
+
+import requests_mock
+from airbyte_cdk.test.http import HttpRequest, HttpRequestMatcher, HttpResponse
+
+
+class HttpMocker(contextlib.ContextDecorator):
+    """
+    WARNING: This implementation only works if the lib used to perform HTTP requests is `requests`
+    """
+
+    def __init__(self) -> None:
+        self._mocker = requests_mock.Mocker()
+        self._matchers: List[HttpRequestMatcher] = []
+
+    def __enter__(self) -> "HttpMocker":
+        self._mocker.__enter__()
+        return self
+
+    def __exit__(self, exc_type: Optional[BaseException], exc_val: Optional[BaseException], exc_tb: Optional[TracebackType]) -> None:
+        self._mocker.__exit__(exc_type, exc_val, exc_tb)
+
+    def _validate_all_matchers_called(self) -> None:
+        for matcher in self._matchers:
+            if not matcher.was_called():
+                raise ValueError(f"Expected all matchers to be called at least once but `{matcher}` wasn't")
+
+    def get(self, request: HttpRequest, response: HttpResponse) -> None:
+        """
+        WARNING: Given multiple requests that are not mutually exclusive, the request will match the first one. This can happen in scenarios
+        where the same request is added twice (in which case there will always be an exception because we will never match the second
```

Comment:
fine for the first iteration but I can imagine scenarios where we would want to map the same request to different responses. eg for testing handling of 5XX errors. Maybe the method should accept a list of responses and the matcher would iterate over them?

Reply:
yes makes sense to me!